Past grants improvements

mattandrews opened the pull request:
Quite a few tweaks in here:

Error handling:

![image](https://user-images.githubusercontent.com/394376/46531046-bc516a00-c893-11e8-866b-7dacaa03cf38.png)

Amount awarded facet now comes from the API:
![image](https://user-images.githubusercontent.com/394376/46531065-ca9f8600-c893-11e8-88c0-a0d5d3d52b8e.png)

Typed search works: 
![past-grants-search](https://user-images.githubusercontent.com/394376/46531094-e2770a00-c893-11e8-974c-dbee8d6d0647.gif)

Pagination and sorting links are now dynamic:
![past-grants-search-sort](https://user-images.githubusercontent.com/394376/46531129-f884ca80-c893-11e8-999b-39a4f3c30c70.gif)

Individual filters can be cleared:
![past-grants-search-clear](https://user-images.githubusercontent.com/394376/46531167-210cc480-c894-11e8-9828-4d494131c27b.gif)

And noscript support offers all the same fields (just not dynamic). Bingo!

mattandrews commented on assets/js/vue-apps/grant-filters.js:
```diff
@@ -10,80 +11,145 @@ function init() {
+                const type = this.sort.type;
+                const dir = this.sort.direction;
+                return {
+                    'sort-controls__item--active': type === sortKey,
```
It might be a bit ugly having CSS classnames in JS, but on the other hand it looked quite ugly having chained JS logic in the HTML, so...

davidrapson replied:
In which case let's go for a "state" class like `is-active`, `is-descending`, `is-ascending`, rather than using component-specific classes.

mattandrews replied:
Done!

mattandrews commented on controllers/past-grants/views/index.njk:
```diff
@@ -37,15 +49,22 @@
                                value="{{ queryParams.q }}"
                                placeholder="Enter a search term, location, or postcode"
-                               v-on:change="filterResults"
+                               autocomplete="nope"
```
This is weird but [recommended by MDN](https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion).

davidrapson replied:
Love it

davidrapson commented on assets/js/vue-apps/grant-filters.js:
```diff
@@ -10,80 +11,145 @@ function init() {
                 defaultFilters,
-                filters
+                sort: Object.assign({}, existingSort),
+                facets: Object.assign({}, existingFacets),
```
Do we need to assign when there is only one object?

mattandrews replied:
This was to clone them rather than extend anything, have been bitten in the past by mutation when just doing plain `facets = existingFacets`.

davidrapson commented on assets/js/vue-apps/grant-filters.js:
```diff
@@ -10,80 +11,145 @@ function init() {
+                handler: function() {
+                    this.filterResults();
+                },
+                deep: true
```
But how deep are you really :neckbeard:

mattandrews replied:
2deep4u

davidrapson commented on assets/sass/utilities/_utilities-display.scss:
```diff
@@ -114,6 +114,10 @@
     @include visually-hidden();
 }
 
+.u-invisible {
```
Nothing to see here

mattandrews commented:
Made a few further tweaks:

Added a constituency filter (for the MPs):
![image](https://user-images.githubusercontent.com/394376/46535442-4d7c0d00-c8a3-11e8-9881-6af93b4dc890.png)

Tweaked the detail page:
![image](https://user-images.githubusercontent.com/394376/46535452-5836a200-c8a3-11e8-8d0d-049104cb421d.png)

Added a read-only filter for recipient:
![image](https://user-images.githubusercontent.com/394376/46535467-65ec2780-c8a3-11e8-92c4-b333d2a761f1.png)

Added a read-only filter for org type:
![image](https://user-images.githubusercontent.com/394376/46535483-6edcf900-c8a3-11e8-93c5-abfc70e04c71.png)

(these two are read-only because we don't want a dropdown facet of recipients, and possibly org type is a bit pointless until we get more fine-grained results, but for now you can click through from the detail page to either of those things)

davidrapson commented on assets/js/vue-apps/grant-filters.js:
```diff
@@ -1,5 +1,6 @@
 import $ from 'jquery';
 import Vue from 'vue';
+import _ from 'lodash';
```
Import the specific method otherwise this will pull in all of lodash

davidrapson commented on assets/js/vue-apps/grant-filters.js:
```diff
@@ -10,80 +11,146 @@ function init() {
+                                    this.facets = response.facets;
+                                    this.updateUrl();
+                                } else {
+                                    if (response.error.error.error && response.error.error.error.code) {
```
error
error?
error!
